server/builtin/install_consent.go

```diff
@@ -49,15 +50,51 @@ func (a *builtinApp) installConsentSubmit(creq apps.CallRequest) apps.CallRespon
 
 	_, out, err := a.proxy.InstallApp(
 		proxy.NewIncomingFromContext(creq.Context),
-		creq.Context, appID, true, secret)
+		creq.Context, appID, deployType, true, secret)
 	if err != nil {
 		return apps.NewErrorCallResponse(errors.Wrap(err, "failed to install App"))
 	}
 
 	return mdResponse(out)
 }
 
+func (a *builtinApp) newConsentDeployTypeField(m apps.Manifest, creq apps.CallRequest) (field apps.Field, selected apps.SelectOption) {
+	opts := []apps.SelectOption{}
+	for _, deployType := range m.DeployTypes() {
+		_, canUse := a.proxy.CanDeploy(deployType)
+		if canUse {
+			opts = append(opts, apps.SelectOption{
+				Label: deployType.String(),
+				Value: string(deployType),
+			})
+		}
+	}
+
+	dtype := apps.DeployType(creq.GetValue(fDeployType, ""))
+	defaultValue := apps.SelectOption{
+		Label: dtype.String(),
+		Value: string(dtype),
+	}
+	if len(opts) == 1 {
+		defaultValue = opts[0]
+	}
+
+	return apps.Field{
+		Name:                fDeployType,
+		Type:                apps.FieldTypeStaticSelect,
+		IsRequired:          true,
+		Description:         "Select how the App will be accessed.",
```

Comment:
Could you elaborate the use case we are solving here? If an admin fetches the manifest via e.g. HTTP, in which case would there also be the same app running in e.g. AWS?

Reply:
I changed this ultimately. Now `appsctl` installs the manifest locally and it contains only the "deployed" options. `apps install listed` installs and displays the available (deployed) options, will finalize in Part 4, need some more `appsctl` work.